File: September-LeetCoding-Challenge/07-Word-Pattern/Word-Pattern.cs
public class Solution {
    public bool WordPattern(string pattern, string str) {
            var words = str.Split(" ");

            if (pattern.Length != words.Length) return false;

            var dict = new Dictionary<char, string>();

            for (int i = 0; i < words.Length; i++)
            {
                char c = pattern[i]; //Positionning char

                if (dict.ContainsKey(c))
                {
                    if (dict[c] != words[i])
                        return false;
                }
                else
                {
                    if (dict.ContainsValue(words[i]))
                        return false;

                    dict.Add(c, words[i]);
                }
            }

            return true;
    }
}

File: September-LeetCoding-Challenge/07-Word-Pattern/Word-Pattern.rb
def word_pattern(pattern, str)
    arr = str.split
    return false if pattern.length != arr.length
    l = pattern.length
    h = {}
    out = true
    0.upto(l-1) do |n|

        if h.key?(pattern[n])
            return out = false if h[pattern[n]] != arr[n]
        else
            return false if h.values.include?(arr[n])
            h.merge!(pattern[n] => arr[n])
        end
    end
    out
end